rust_src/src/json_rpc.rs

```diff
@@ -0,0 +1,63 @@
+use remacs_macros::lisp_fn;
+use lazy_static::lazy_static;
+use std::thread;
+use crate::lisp::LispObject;
+use crate::remacs_sys::{XPROCESS};
+
+use std::{
+    fs::File,
+    io::{self, Write, Read, BufReader},
+    os::unix::io::{FromRawFd, IntoRawFd},
+};
+
+#[repr(u32)]
+enum ProcessHandles {
+    SUBPROCESS_STDIN = 0,
+    WRITE_TO_SUBPROCESS = 1,
+    READ_FROM_SUBPROCESS = 2,
+    SUBPROCESS_STDOUT = 3,
+    READ_FROM_EXEC_MONITOR = 4,
+    EXEC_MONITOR_OUTPUT = 5,
+}
+
+#[lisp_fn]
+pub fn thread_path(proc: LispObject) -> bool {
+    let proc_raw = unsafe { XPROCESS(proc) };
```

Comment:
I'm trying to find the code that allows `LispProcessRef` as argument to avoid this line.